src/Core/Parser/Patterns.php

```diff
@@ -229,22 +229,25 @@
      *
      */
     static public $SPLIT_PATTERN_SHORTHANDSYNTAX_ARRAY_PARTS = '/
-		(?P<ArrayPart>                                             # Start sub-match
-			(?P<Key>                                               # The keys of the array
-				[a-zA-Z0-9\\-_]+                                   # Unquoted
-				|"(?:\\\"|[^"])+"                                  # Double quoted
-				|\'(?:\\\\\'|[^\'])+\'                             # Single quoted
+		(?P<ArrayPart>                                                      # Start sub-match
+			(?P<Key>                                                        # The keys of the array
+				 [a-zA-Z0-9_-]+                                             # Unquoted
+				|"(?:\\\\"|[^"])+"                                          # Double quoted
+				|\'(?:\\\\\'|[^\'])+\'                                      # Single quoted
 			)
-			\s*:\s*                                                # Key|Value delimiter :
-			(?:                                                    # Possible value options:
-				(?P<QuotedString>                                  # Quoted string
-					(?:"(?:\\\"|[^"])*")
-					|(?:\'(?:\\\\\'|[^\'])*\')
+			\\s*:\\s*                                                       # Key|Value delimiter :
+			(?:                                                             # Possible value options:
+				(?P<QuotedString>                                          # Quoted string
+					 "(?:\\\\"|[^"])*"
+					|\'(?:\\\\\'|[^\'])*\'
 				)
-				|(?P<VariableIdentifier>[a-zA-Z0-9\-_.]*[a-zA-Z]+)  # variable identifiers must contain letters (otherwise they are hardcoded numbers)
-				|(?P<Number>[0-9.]+)                               # Number
-				|{\s*(?P<Subarray>(?:(?P>ArrayPart)\s*,?\s*)+)\s*} # Another sub-array
-			)                                                      # END possible value options
-		)                                                          # End array part sub-match
+				|(?P<VariableIdentifier>
+					(?:(?=[^,{}\.]*[a-zA-Z])[a-zA-Z0-9_-]*)                        # variable identifiers must contain letters (otherwise they are hardcoded numbers)
+					(?:\\.[a-zA-Z0-9_-]+)*                                  # but subsequent access of keys and properties may be only numbers
+				)
+				|(?P<Number>[0-9]+(?:\\.[0-9]+)?)                           # Number
+				|\\{\\s*(?P<Subarray>(?:(?P>ArrayPart)\\s*,?\\s*)+)\\s*\\}  # Another sub-array
+			)                                                               # END possible value options
+		)\\s*(?=\\z|,|\\})                                                           # End array part sub-match
```

Comment:
The ArrayPart is used recursively. Hence at least comma and brace need to be covered. I just added the end of string anchor to ensure we match for an end in any case. This was totally missing before which leads to false positives to be accepted. `2.2bar` would be parially excepted then